src/core/gps/qgsqtlocationconnection.cpp

```diff
@@ -70,6 +70,19 @@ void QgsQtLocationConnection::parseData()
       mLastGPSInformation.direction = mInfo.attribute( QGeoPositionInfo::Direction );
       mLastGPSInformation.utcDateTime = mInfo.timestamp();
       mLastGPSInformation.fixType = mInfo.coordinate().type() + 1;
+      switch ( mInfo.coordinate().type() )
+      {
+        case QGeoCoordinate::InvalidCoordinate:
+          mLastGPSInformation.mConstellationFixStatus[ Qgis::GnssConstellation::Unknown ] = Qgis::GpsFixStatus::NoFix;
```

Comment:
For the record, we could greatly improve QGIS' Qt location connection class by implementing QGeoSatelliteInfoSource (https://doc.qt.io/qt-6/qgeosatelliteinfosource.html). In Qt6, it even has the full array of GNSS services out there (https://doc.qt.io/qt-6/qgeosatelliteinfo.html#SatelliteSystem-enum).

_Definitively not blocking any of this_, just dropping in a note :) QField's implementation is here (https://github.com/opengisch/QField/blob/master/src/core/positioning/internalgnssreceiver.cpp#L43).